In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
%matplotlib inline
import warnings
warnings.filterwarnings("ignore")

# 1) Overview of the Dataset

In [2]:
df = pd.read_csv("data/stud.csv")

In [3]:
df.head()

gender race_ethnicity parental_level_of_education         lunch  \
0  female        group B           bachelor's degree      standard   
1  female        group C                some college      standard   
2  female        group B             master's degree      standard   
3    male        group A          associate's degree  free/reduced   
4    male        group C                some college      standard   

  test_preparation_course  math_score  reading_score  writing_score  
0                    none          72             72             74  
1               completed          69             90             88  
2                    none          90             95             93  
3                    none          47             57             44  
4                    none          76             78             75

## 1.1) Shape of the dataset

In [4]:
df.shape

(1000, 8)

## 1.2) Dataset Information

- gender: sex of students -> Male/Female
- race/ethnicity: ethnicity of students -> Group A, B, C, D, E
- parental level of education: parents' final education -> Bachelor's degree, some college, master's degree, associate's degree, high school
- lunch: having lunch before test -> standard or free/reduced
- test preparation course: completed or not before a test
- math score
- reading score
- writing score

# 2) Data Checks to perform

- Check missing values
- Check duplicates
- Check the number of unique values of each column
- Check statistics of data set
- Check various categories present in the different categorical columns

## 2.1) Check for missing values

In [5]:
df.isna().sum()

gender                         0
race_ethnicity                 0
parental_level_of_education    0
lunch                          0
test_preparation_course        0
math_score                     0
reading_score                  0
writing_score                  0
dtype: int64

There are no missing values in the dataset

## 2.2) Check duplicates

In [6]:
df.duplicated().sum()

np.int64(0)

There are no duplicates values in the dataset

## 2.3)Check data types

In [7]:
# check Null and dtypes
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race_ethnicity               1000 non-null   object
 2   parental_level_of_education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test_preparation_course      1000 non-null   object
 5   math_score                   1000 non-null   int64 
 6   reading_score                1000 non-null   int64 
 7   writing_score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


## 2.4) Checking the number of unique values of each column

In [9]:
df.nunique()

gender                          2
race_ethnicity                  5
parental_level_of_education     6
lunch                           2
test_preparation_course         2
math_score                     81
reading_score                  72
writing_score                  77
dtype: int64

## 2.5) Check statistics of the dataset

In [10]:
df.describe()

math_score  reading_score  writing_score
count  1000.00000    1000.000000    1000.000000
mean     66.08900      69.169000      68.054000
std      15.16308      14.600192      15.195657
min       0.00000      17.000000      10.000000
25%      57.00000      59.000000      57.750000
50%      66.00000      70.000000      69.000000
75%      77.00000      79.000000      79.000000
max     100.00000     100.000000     100.000000

### Insight

- From the above description of numerical data,all means are very close to each other - between 60 and 68.05.
- All standard deviations are also close - between 14.6 and 15.19.
- While there is a minimum score 0 for math, for writing the minimum is much higher = 10 and for reading even higher = 17.

## 2.6) Exploring Data